src/pudl/metadata/classes.py

```diff
@@ -529,6 +536,22 @@ def from_id(cls, x: str) -> 'Encoder':
         """Construct an Encoder based on `Resource.name` of a coding table."""
         return cls(**cls.dict_from_id(x))
 
+    @classmethod
+    def from_code_id(cls, x: str) -> 'Encoder':
+        """Construct an Encoder based on looking up the name of a coding table directly in the codes metadata."""
+        return cls(**copy.deepcopy(CODE_METADATA[x]), name=x)
+
+    def to_rst(self, abs_csv_dir_path: DirectoryPath, rel_csv_dir_path: DirectoryPath, is_header: Bool) -> String:
+        """Ouput dataframe to a csv for use in jinja template. Then output to an RST file."""
+        abs_csv_filepath = Path(abs_csv_dir_path, f"{self.name}.csv")
+        self.df.to_csv(abs_csv_filepath, index=False)
+        rel_csv_filepath = Path(rel_csv_dir_path, f"{self.name}.csv")
+        description = RESOURCE_METADATA[self.name]["description"]
```

Comment:
Instead of creating these single-use variables which are only used a single time in `template.render()` below, I find it easier to read what's happening with the values themselves as arguments. It saves one step of mental redirection. E.g.

```py
self.df.to_csv(
    abs_csv_dir_path / f"{self.name}.csv",
    index=False,
)

return template.render(
    Encoder=self,
    description=RESOURCE_METADATA[self.name]["description"],
    csv_filepath=rel_csv_dir_path / f"{self.name}.csv",
    is_header=is_header
)
```